### DISCUSSION TITLE
Save additional outputs from likelihood calls

### DISCUSSION THREAD

**Discussion by j-dr** [blackjax-devs/blackjax#715] [Q&A]
I'm currently using the NUTS sampler for an analysis that I'm conducting and there are a number of variables that are calculated in my likelihood code that I would like to save as outputs in addition to the log density and posterior samples. Is there a straight-forward way to do this with blackjax? Thanks!

**Comment by junpenglao** [comment]
🤖 *This reply was drafted with Claude and reviewed by me (@junpenglao).*

There's no aux-output channel inside the kernel, but two options work well: (1) compute the extra quantities from the saved positions in a `vmap` after sampling; or (2) pass a `transform` to `blackjax.util.run_inference_algorithm` to store derived values per step. If you use NumPyro, `postprocess_fn` / deterministics give you these directly (see #792).
